### PR TITLE
users.update [PRO-1749]

### PR DESCRIPTION
**Pull request by adognameddog**
For now we only allow the updating of the internal hourly cost by account administrators. In the (near) future, this endpoint can be expanded so it allows the updating of other user properties.

### CODE REVIEW

File: src/01-general/users.apib [hunk 1]
```diff
@@ -120,3 +120,17 @@ Get details for a single user.
...
+Update a single user. For now, only internal_hourly_rate is available for update.
+Only account administrators can update the internal_hourly_rate.
+
++ Request (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + id: `b5094b3f-bb7a-0391-b01b-e709773f3509` (string)
+        + internal_hourly_cost (Money)
+
++ Response 204 (application/json;charset=utf-8)
```
Comment (joaoinacio) [inline src/01-general/users.apib]: Q/B: I think `charset=utf-8` has been removed in master?

Comment (joaoinacio) [inline src/01-general/users.apib]:
```suggestion
+ Response 204 (application/json)
```

**Review by ketlane — DISMISSED** [review]

**Review by ArnaudWeyts — DISMISSED** [review]

File: apiary.apib [hunk 2]
```diff
@@ -625,6 +625,20 @@ Get details for a single user.
...
+### users.update [POST /users.update]
+
+Update a single user. For now, only internal_hourly_rate is available for update.
+Only account administrators can update the internal_hourly_rate.
+
++ Request (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + id: `b5094b3f-bb7a-0391-b01b-e709773f3509` (string)
+        + internal_hourly_cost (Money)
```
Comment (jenssegers) [inline apiary.apib]: PP: Add a comment next to the property that only account admins can update it.

**Review by ketlane — DISMISSED** [review]

**Review by jenssegers — DISMISSED** [review]

**Comment by jenssegers** [comment]
@joaoinacio Is right about the content type though!

File: src/01-general/users.apib [hunk 3]
```diff
@@ -120,3 +120,16 @@ Get details for a single user.
                     + tr-TR
             + function: `Sales` (string)
             + time_zone: `Europe/Brussels` (string)
+
+
+### users.update [POST /users.update]
+
+Update a single user. For now, only internal_hourly_rate is available for update.
+
++ Request (application/json;charset=utf-8)
```
Comment (jenssegers) [inline src/01-general/users.apib:140]:
```suggestion
+ Request (application/json)
```

File: apiary.apib [hunk 4]
```diff
@@ -625,6 +625,19 @@ Get details for a single user.
             + function: `Sales` (string)
             + time_zone: `Europe/Brussels` (string)
 
+
+### users.update [POST /users.update]
+
+Update a single user. For now, only internal_hourly_rate is available for update.
+
++ Request (application/json;charset=utf-8)
```
Comment (jenssegers) [inline apiary.apib:845]:
```suggestion
+ Request (application/json)
```

**Review by stivni — DISMISSED** [review]
@adognameddog why don't we merge this (and all other profit related additions) to an integration-branch for profit (like for workorders and deals), so that it is only released when we are happy about the totality, but this PR is of our plate?

**Comment by stivni** [comment]
@adognameddog @jenssegers 

I recall having a discussion about the profile.update endpoint and users.update...

Where we said that we don't want to mix things that you can only update yourself with things that you can't update yourself... so there was a `profile.update` at some point.

But the suggestion was to story this in an `HR` related endpoint.
Is this something we can/should do before merging this one?

**Comment by adognameddog** [comment]
I'm going to move this to our internal api documentation

**Pull request closed without merging**